### PR TITLE
Jlc/change staff admin msg pipeline

### PR DESCRIPTION
**Pull request by johanseto**
## Description

[refactor: change pipelien auth exception to 403](https://github.com/eduNEXT/eox-nelp/pull/113/commits/40ec5409073ea8f03539f3e71632142affc174ca)
## Testing instructions
Use a Saml flow using a staff and admin user. 
Check the authentication msg.
### Before
Peek 2023-12-11 16-29

#### English
Peek 2023-12-11 16-35

#### Arabic
Peek 2023-12-11 16-37

## Additional information

**Jira story**
https://edunext.atlassian.net/browse/FUTUREX-641?atlOrigin=eyJpIjoiNmI0MTAyNzBjMjM4NDZiNjgyMTk4MDM5MjM3ZjViODUiLCJwIjoiaiJ9

## Checklist for Merge

- [x] Tested in a remote environment
- [ ] Updated documentation
- [ ] Rebased master/main
- [ ] Squashed commits

### CODE REVIEW

File: eox_nelp/third_party_auth/tests/test_pipeline.py [hunk 1]
```diff
@@ -24,15 +25,17 @@ def setUp(self):  # pylint: disable=invalid-name
...
             - Return None.
         """
-        backend = Mock()
-
-        pipe_output = safer_associate_username_by_uid(backend, self.details, self.response, user=self.user)
+        pipe_output = safer_associate_username_by_uid(
+            self.request, self.backend, self.details, self.response, user=self.user,
+        )
 
         self.assertEqual(None, pipe_output)
```
Comment (andrey-canon) [inline eox_nelp/third_party_auth/tests/test_pipeline.py:40]:
```suggestion
        self.assertIsNone(None, pipe_output)
```

File: eox_nelp/third_party_auth/tests/test_pipeline.py [hunk 2]
```diff
@@ -60,17 +63,21 @@ def test_staff_user_raise_exc(self, user_kwargs):
         """Test the pipeline method try to match with uid, the user with matched username exists
         but is staff or superuser.
         Expected behavior:
-            - Raise EoxNelpAuthException exception
+            - The pipe return an HttpResponseForbidden object.
```
Comment (andrey-canon) [inline eox_nelp/third_party_auth/tests/test_pipeline.py:66]:
```suggestion
            - The pipeline returns an HttpResponseForbidden object.
```

File: eox_nelp/third_party_auth/tests/test_pipeline.py [hunk 3]
```diff
@@ -60,17 +63,21 @@ def test_staff_user_raise_exc(self, user_kwargs):
...
-        backend.strategy.storage.user.get_user.return_value = past_user
 
-        self.assertRaises(EoxNelpAuthException, safer_associate_username_by_uid, backend, self.details, self.response)
-        backend.strategy.storage.user.get_user.assert_called_with(username=test_uid)
+        self.backend.get_idp.return_value.get_user_permanent_id.return_value = test_uid
+        self.backend.strategy.storage.user.get_user.return_value = past_user
+
+        pipe_output = safer_associate_username_by_uid(self.request, self.backend, self.details, self.response)
+
+        self.assertTrue(isinstance(pipe_output, HttpResponseForbidden))
```
Comment (andrey-canon) [inline eox_nelp/third_party_auth/tests/test_pipeline.py:78]:
```suggestion
        self.assertIsInstance(isinstance(pipe_output, HttpResponseForbidden))
```

File: eox_nelp/third_party_auth/pipeline.py [hunk 4]
```diff
@@ -94,8 +96,18 @@ def safer_associate_username_by_uid(  # pylint: disable=unused-argument
...
     if user_match.is_staff or user_match.is_superuser:
-        raise EoxNelpAuthException(backend, "It is not allowed to auto associate staff or admin users")
-
+        return HttpResponseForbidden(
+            edxmako.shortcuts.render_to_string(
+                "static_templates/server-error.html",
+                {
+                    "page_header": _("It is not allowed to auto associate staff or admin users"),
+                    "page_content": _(
+                        "You can change permission for that user or try to check the signupsource of the user."
```
Comment (andrey-canon) [inline eox_nelp/third_party_auth/pipeline.py:105]: I'm sure about the relation of the signupsource model here,anyway I prefer something like "You are trying to access with a user that already exists. Please try to authenticate on the site where you registered for the first time or contact support."

**Review by andrey-canon — CHANGES_REQUESTED** [review]

File: eox_nelp/third_party_auth/tests/test_pipeline.py (hunk 2 above)
Comment (johanseto) [reply eox_nelp/third_party_auth/tests/test_pipeline.py:66]: Here I have some doubts I mean pipeline is not the list of pipes ?
and a pipe is only for one related to a function?
image

File: eox_nelp/third_party_auth/pipeline.py (hunk 4 above)
Comment (johanseto) [reply eox_nelp/third_party_auth/pipeline.py]: [a09575e](https://github.com/eduNEXT/eox-nelp/pull/113/commits/a09575e7c82931e92c6927f28575a7f32fa16048)
image

Comment (johanseto) [reply eox_nelp/third_party_auth/pipeline.py]: image

File: eox_nelp/third_party_auth/tests/test_pipeline.py (hunk 2 above)
Comment (andrey-canon) [reply eox_nelp/third_party_auth/tests/test_pipeline.py:66]: makes sense, however it's the first time that I see that someone uses `pipe` to refer to a single process, that make s me think about the Unix pipe symbol |

File: eox_nelp/third_party_auth/tests/test_pipeline.py (hunk 3 above)
Comment (johanseto) [reply eox_nelp/third_party_auth/tests/test_pipeline.py]: [15ff106](https://github.com/eduNEXT/eox-nelp/pull/113/commits/15ff1069a9c083975dccb28c243cc42dcf8c3ad5)

File: eox_nelp/third_party_auth/tests/test_pipeline.py (hunk 2 above)
Comment (johanseto) [reply eox_nelp/third_party_auth/tests/test_pipeline.py:66]: I thing that could be used in that way. The unix pipe operator is used to connect, but here the pipes also changes the process in the interconnection xD.
_A pipe is a method used to pass information from one program process to another. Unlike other types of interprocess communication, a pipe only offers one-way communication by passing a parameter or output from one process to another._
https://www.techopedia.com/definition/3410/pipe
https://www.geeksforgeeks.org/pipe-system-call/

Comment (andrey-canon) [reply eox_nelp/third_party_auth/tests/test_pipeline.py:66]: were you able to find any reference where someone call this pipe ? I think this make sense but that looks like a literal translation that nobody use

Comment (andrey-canon) [reply eox_nelp/third_party_auth/tests/test_pipeline.py:66]: anyway I don't want to get into semantic discussions IMO pipe is not a common term and I prefer something like pipeline method or function but it's up to you

**Review by andrey-canon — APPROVED** [review]

**Pull request merged**